File: packages/events/python/lockin_events/streams.py
from typing import Final

STREAM_TASKS: Final[str] = "events:tasks"
STREAM_SIGNALS: Final[str] = "events:signals"
STREAM_SCHEDULE: Final[str] = "events:schedule"
STREAM_AGENT: Final[str] = "events:agent"

EVENT_TYPE_TO_STREAM: Final[dict[str, str]] = {
    "task.created": STREAM_TASKS,
    "task.scheduled": STREAM_SCHEDULE,
    "task.accepted": STREAM_SCHEDULE,
    "task.rejected": STREAM_SCHEDULE,
    "task.modified": STREAM_TASKS,
    "task.completed": STREAM_TASKS,
    "mood.logged": STREAM_SIGNALS,
    "energy.logged": STREAM_SIGNALS,
    "schedule.explained": STREAM_AGENT,
}

File: packages/events/src/streams.ts
export const STREAM_NAMES = {
  TASKS: "events:tasks",
  SIGNALS: "events:signals",
  SCHEDULE: "events:schedule",
  AGENT: "events:agent",
} as const;

export type StreamName = (typeof STREAM_NAMES)[keyof typeof STREAM_NAMES];

export const EVENT_TYPE_TO_STREAM: Record<string, StreamName> = {
  "task.created": STREAM_NAMES.TASKS,
  "task.scheduled": STREAM_NAMES.SCHEDULE,
  "task.accepted": STREAM_NAMES.SCHEDULE,
  "task.rejected": STREAM_NAMES.SCHEDULE,
  "task.modified": STREAM_NAMES.TASKS,
  "task.completed": STREAM_NAMES.TASKS,
  "mood.logged": STREAM_NAMES.SIGNALS,
  "energy.logged": STREAM_NAMES.SIGNALS,
  "schedule.explained": STREAM_NAMES.AGENT,
};
